druid/examples/image.rs

```diff
@@ -12,44 +12,46 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! This example shows how to draw an png image.
+//! This example shows how to draw an image. Using the Image feature.
+//! Diferent image formats can be "unlocked" by adding their
+//! corresponding feature.
 
-use druid::{
-    widget::{FillStrat, Flex, Image, WidgetExt},
-    AppLauncher, Color, ImageBuf, Widget, WindowDesc,
-};
+use druid::piet::InterpolationMode;
+use druid::widget::prelude::*;
+use druid::widget::{FillStrat, Flex, Image, WidgetExt};
+use druid::{AppLauncher, Color, ImageBuf, WindowDesc};
 
 pub fn main() {
     let main_window = WindowDesc::new(ui_builder);
-    let data = 0_u32;
     AppLauncher::with_window(main_window)
         .use_simple_logger()
-        .launch(data)
+        .launch(0)
         .expect("launch failed");
 }
 
 fn ui_builder() -> impl Widget<u32> {
     let png_data = ImageBuf::from_data(include_bytes!("./assets/PicWithAlpha.png")).unwrap();
 
-    let mut col = Flex::column();
-
-    col.add_flex_child(
-        Image::new(png_data.clone())
-            .border(Color::WHITE, 1.0)
-            .fix_width(100.0)
-            .center(),
-        1.0,
-    );
-
-    /*
-    // If you want to change the fill stratagy you can but you need the widget to be mut
-    let mut otherimage = Image::new(png_data);
-    otherimage.set_fill(FillStrat::FitWidth);
-    */
-
-    let otherimage = Image::new(png_data)
-        .fill_mode(FillStrat::FitWidth)
-        .border(Color::WHITE, 1.0);
-    col.add_flex_child(otherimage, 1.0);
-    col
+    // We create 2 images, one not having any modifications and the other
+    // is set to a fixed width, a fill strategy and an interpolation mode.
+    // You can see how this affects the final result. You can play with the
+    // Interpolation mode to see hwo this affects things.
+    // Note that this image is already anti-aliased so NearestNeighbor looks
+    // weird
```

Comment:
```suggestion
    // We create two images, one not having any modifications and the other
    // forced to a fixed width, a fill strategy and an interpolation mode.
    // You can see how this affects the final result. You can play with the
    // interpolation mode to see how this affects things. (Note that this image
    // is already anti-aliased so NearestNeighbor looks weird.)
```

Note on this: we use the same `FillStrat` in both examples, so I'm not sure it's correct to say that the one has "no modifications"? What do you think?

Reply:
yeah that makes more sense, thanks for catching that!

Reply:
I just tested and the fillstrat doesn't actually matter when there is no fixed width.